### DISCUSSION TITLE
Setting up modern monorepo with browser and node targets [v2]

### DISCUSSION THREAD

**Discussion by xzilja** [parcel-bundler/parcel#4802] [General]
Sorry if this discussion ends up verbose, but this is my first experience with parcel and I am trying to figure out how to use it for a modern monorepo setup that targets browser / nodejs and exposes some shared modules.

1. Assuming we are starting with yarn workspaces and our structure is similar to below:
```
packages/
     app/
          index.tsx
          index.html
          package.json
     core/
          utilities.ts
     server/
           index.ts
package.json
tsconfig.json
```

2. What I did so far is add parcel to root package json and added `start` and `build` commands in `app/package.json`, so far so good. I then proceeded to set up aliases for these packages inside root `package.json` i.e.

```json
{
  "alias": {
    "core": "packages/core/**/*",
    "app": "packages/app/**/*",
    "server": "packages/server/**/*"
  },
}
```

However when I tried to `import utilities from "core/utilities"` inside my `app` package, I got an error saying that this failed to resolve, so I assume I did something wrong here? I am also unable to use `~/packages/core/utilities`, as it resolves incorrectly as well.

3.  I paused with alias, and went on with reading the documentation. I think I need to specify `target` for each of my packages now, but I am not sure if I shold do it in individual package json files or in root one, furthermore I want  to somehow make it so that my app only produces code necessary to run it in latest chrome / safari and my server needs to produce bundles for node v14. In terms of where I want  too build this, a `dist` folder for each package (i.e. `packages/app/dist`) will work. Now how would I approach this to make it more managable?

Would appreciate any examples / deep dives into this if possible. Need to understand how to set things like targets / engines / browserlist here correctly and produce modern bundles, while also utilising features like alias for better dev experience

**Comment by jamesvclements** [comment]
@IljaDaderko did you end up figuring out this set-up, and/or would you recommend a different route in 2022?
